modules/RtcStats/RtcStats.js

```diff
@@ -0,0 +1,138 @@
+    setMeetingF;
+    meetingFqnSet = new Promise(r => {
+        this.setMeetingF = r;
+    });
+
+    /**
+     * Initializes the underlying rtcstats node moduke.
+     */
+    init(options) {
+        const { rtcstatsUseLibJitsi } = options.analytics;
+
+        if (rtcstatsUseLibJitsi) {
+            if (!this.initialized) {
+                this.meetingFqnSet.then(() => {
+                    logger.info(`meetingFqn is set to ${this.meetingFqn}, now initializing rtcstats`);
+
+                    const { rtcstatsEndpoint, rtcstatsUseLegacy, rtcstatsPollInterval, rtcstatsSendSdp }
+                        = options.analytics;
+
+                    const traceOptions = {
+                        endpoint: rtcstatsEndpoint,
+                        meetingFqn: this.meetingFqn,
+                        onCloseCallback: this.handleTraceWSClose.bind(this),
+                        useLegacy: rtcstatsUseLegacy
+                    };
+
+                    const rtcstatsOptions = {
+                        connectionFilter,
+                        pollInterval: rtcstatsPollInterval,
+                        useLegacy: rtcstatsUseLegacy,
+                        sendSdp: rtcstatsSendSdp,
+                        eventCallback: this.handleRtcStatsEvent.bind(this)
+                    };
+
+                    this.trace = traceInit(traceOptions);
+
+                    rtcstatsInit(this.trace, rtcstatsOptions);
+
+                    this.setTrace(this.trace);
+
+                    this.initialized = true;
+                });
+            }
+        } else {
+            logger.info('RtcStats configured to not use in lib-jitsi-meet');
+        }
+    }
+
+    /**
+     * Returns the handle to RtcStats tracer.
+     *
+     * @returns {Promise}
+     */
+    getTrace() {
+        return this.traceConfigured.then(t => t);
+    }
+
+    /**
+     * Set the meetingFqn value needed for initialzing rtcstats.
+     *
+     * @param {String} - the meeting fqn.
+     * @returns {void}
+     */
+    setMeetingFqn(config) {
+        const { meetingName, eventCallback, onCloseCallback } = config;
+
+        this.eventCallback = eventCallback;
+        this.closeCallback = onCloseCallback;
+
+        logger.info(`Setting meetingFqn to ${meetingName}`);
+        this.meetingFqn = meetingName;
+        this.setMeetingF();
+    }
+
+    /**
+     * RTCStats client can notify the App of any PeerConnection related events.
+     *
+     * @param {Object} event - Rtcstats event.
+     * @returns {void}
+     */
+    handleRtcStatsEvent(event) {
+        logger.info(`RtcStats received event ${JSON.stringify(event)}`);
+        if (typeof this.eventCallback === 'function') {
+            this.eventCallback(event);
+        }
+    }
+
+    /**
+     * Handle WS close.
+     *
+     * @param {Object} closeEvent - Event sent by ws onclose.
+     * @returns {void}
+     */
+    handleTraceWSClose(closeEvent) {
+        logger.warn('RtcStats trace WS closed', closeEvent);
```

Comment:
Reconnects happen inside rtcstats, right? If so, shouldn't there be a way to notify users about when we re-connected?

Reply:
Interesting question.
It looks like the client side reconnection PR got merged https://github.com/jitsi/rtcstats/pull/28 and it just internally buffers all messages which didn't get send to the server successfully. But it is a good question why we now only have a notification for close, but not for reconnect. If the rtcstats library handles everything internally even the disconnect should not get exposed. 
@andrei-gavrilescu do you have any thoughts on how to proceed here? Should we removed the trace WS close callback since everything get handled internally? Or should we expose the reconnect event as well (but then we would need to make it clear that still everything in between got buffer)?